### ISSUE TITLE
Shell error when attempting to kitchen converge freebsd instance (if not created first)

### ISSUE THREAD

**Issue by tbunnyman** [test-kitchen/test-kitchen#712]
* Mac OS 10.10
* chef-dk 0.5.1
* kitchen 1.4.0
* vagrant driver
* chef_zero provisioner

Attempting to kitchen converge freebsd-10.1 or freebsd-9.3 causes the following error
```
STDERR: The configured shell (config.ssh.shell) is invalid and unable
to properly execute commands. The most common cause for this is
using a shell that is unavailable on the system. Please verify
you're using the full path to the shell and that the shell is
executable by the SSH user.
```

If I first run a `kitchen create freebsd-101` this completes fine, and then I can run `converge`, `verify`, and whatever else I need to do until the box is destroyed, and then it needs to be `create`ed again before anything can be done on it.

Here is a gist of me demonstrating this issue https://gist.github.com/d5250722d560f19a4a66
Here is the cookbook I am demonstrating with with https://github.com/ChefFurs/fish-shell

I attempted this on two identical macs.

**Comment by Momus** [comment]
@tbunnyman  This error is caused by the fact that BSDs do not use Bash as the default shell, they use good, old-fashioned /bin/sh So when Vagrant tries to connect when it creates, it checks to make sure it uses bash, and, not finding it, raises an exception.  However, it seems like if you are doing anything to an existing Vagrant box, Vagrant _dosen't_ check the shell, and so it does fine.

The fix is to have the following to your `kitchen.yml`   in order that it generate a Vagrantfile that is correct for BSD. 
```ruby
driver:
  name: vagrant
  ssh:
    shell: '"/bin/sh"'
```` 
Notice the single quotes surrounding the double set of quotes around "/bin/sh" : without them, the Vagrantfile does not generate correctly: the string is passed as a stream of characters and the quotes are lost.  With them, your auto-generated Vagrantfile should look like this:
```ruby
Vagrant.configure("2") do |c|
  c.berkshelf.enabled = false if Vagrant.has_plugin?("vagrant-berkshelf")
  c.vm.box = "opscode-freebsd-10.1"
  c.vm.box_url = "https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_freebsd-10.1_chef-provisionerless.box"
  c.vm.hostname = "default-freebsd-101"
  c.ssh.shell = "/bin/sh"
  c.vm.synced_folder ".", "/vagrant", disabled: true
  c.vm.provider :virtualbox do |p|
  end
end
```
and everything should work.

This is a bit of an annoyance, but hopefully there will be a fix.  Maybe after I figure out how to use test-kitchen I'll take a crack at it.

**Comment by flaccid** [comment]
Have noticed now that use of this globally will result in rendering invalid Vagrantfiles for other platforms i.e. `c.ssh.shell = ""/bin/sh""`.

**Comment by flaccid** [comment]
Ok, I think this might be of been fixed with test-kitchen somewhere along the line as its working with single quotes now for example:
```
platforms:
  - name: freebsd-10.1
    attributes:
      system:
        primary_interface: em1
    driver_config:
        ssh:
          shell: '/bin/sh'
        network:
          - ["private_network", {ip: "123.456.789.0"}]
```
This is what I'm doing for each freebsd platform in my `.kitchen.yml`. Good chance this ticket can be closed now imo.

**Comment by flaccid** [comment]
Tested by upgrading from chef-dk 0.6.2 to 0.9.0 on OS X showed as fixed which is actually just kitchen 1.4.0 to 1.4.2 (I assumed fixed in one of the two patches).

**Comment by cheeseplus** [comment]
Closing this as answered and nowadays vagrant plays a bit nicer with FreeBSD
